src/providers/ad/ad_gpo.c

```diff
@@ -4390,15 +4390,18 @@ ad_gpo_get_sd_referral_send(TALLOC_CTX *mem_ctx,
      */
     state->ref_domain = find_domain_by_name(state->host_domain,
                                             lud->lud_host, true);
-    ldap_free_urldesc(lud);
     if (!state->ref_domain) {
         DEBUG(SSSDBG_CRIT_FAILURE,
               "Could not find domain matching [%s]\n",
               lud->lud_host);
+        ldap_free_urldesc(lud);
         ret = EIO;
         goto done;
     }
 
+    ldap_free_urldesc(lud);
```

Comment:
Instead of having the ldap_free_urldesc() freed on both branches, I'd just nullify the lud after the ldap_free_urldesc() in before the if branch.

However, I do **not** have any strong opinion on this.

Reply:
I'm not sure where would I put the free, can you tell me the line number of the if before which you want to free lud? It can't be line 4393, because the lud struct is dereferenced in the if